src/extremeweatherbench/regions.py

```diff
@@ -1,55 +1,84 @@
+import pathlib
 from collections import namedtuple
-from pathlib import Path
 from typing import Any, Type
 
 import geopandas as gpd  # type: ignore[import-untyped]
 import numpy as np
 import regionmask
+import shapely  # type: ignore[import-untyped]
 import xarray as xr
-from shapely import MultiPolygon, Polygon  # type: ignore[import-untyped]
 
 from extremeweatherbench import utils
 
 logger = logging.getLogger(__name__)
 
 
-class Region(ABC):
+class Region(abc.ABC):
     """Base class for different region representations."""
 
     @classmethod
-    @abstractmethod
+    @abc.abstractmethod
     def create_region(cls, *args, **kwargs) -> "Region":
         """Abstract factory method to create a region; subclasses must implement with
         their own, specialized arguments."""
         pass
 
     @property
-    @abstractmethod
+    @abc.abstractmethod
     def geopandas(self) -> gpd.GeoDataFrame:
         """Return representation of this Region as a GeoDataFrame."""
         pass
 
-    def mask(self, dataset: xr.Dataset, drop: bool = False) -> xr.Dataset:
-        """Mask a dataset to the region."""
-        mask = regionmask.mask_geopandas(
-            self.geopandas, dataset.longitude, dataset.latitude
-        )
-        mask_array = ~np.isnan(mask)
-        return dataset.where(mask_array, drop=drop)
-
     @property
     def get_bounding_coordinates(self) -> tuple[Any, ...]:
-        """Get the bounding coordinates of the region."""
+        """Get the bounding coordinates of the region.
+
+        Returns:
+            A named tuple with longitude_min, latitude_min, longitude_max, and
+            latitude_max.
+        """
         return namedtuple(
             "BoundingCoordinates",
             ["longitude_min", "latitude_min", "longitude_max", "latitude_max"],
         )(*self.geopandas.total_bounds)
 
+    def mask(self, dataset: xr.Dataset, drop: bool = False) -> xr.Dataset:
+        """Mask a dataset to the region.
+
+        Args:
+            dataset: The dataset to mask.
+
+        Returns:
+            The subset dataset.
+        """
+        if drop:
+            logger.info(
+                "drop is no longer used and will be removed in a future version."
+            )
+        # If the lats are monotonically decreasing, reverse the slice (max, min)
+        longitude_min, latitude_min, longitude_max, latitude_max = (
+            self.geopandas.total_bounds
+        )
+        latitude_order = dataset.latitude.values.argsort()
+        if latitude_order[0] > latitude_order[-1]:
+            dataset = dataset.sel(
+                latitude=slice(latitude_max, latitude_min),
+                longitude=slice(longitude_min, longitude_max),
+            )
+        else:
+            # If monotonically increasing, slice normally (min, max)
+            dataset = dataset.sel(
+                latitude=slice(latitude_min, latitude_max),
+                longitude=slice(longitude_min, longitude_max),
+            )
+
```

Comment:
This is a good example of where enforcing a standard on the input data makes life simpler. Pick either convention of "latitudes always increase from -90" or "latitudes always decrease from 90", enforce that at the dataset creation level (you can add sanity checks to warn the user that the data is formatted incorrectly right after the data is passed in to the evaluate function), and then eliminate branching code paths like this. You can still keep a sanity check, but do not give the user more than one way to do things.